chat_shell/chat_shell/agents/graph_builder.py

```
@@ -568,6 +646,103 @@ async def stream_tokens(
+                        if has_tool_calls:
+                            # Tool calls present with truncation - raise exception for recovery
+                            add_span_event(
+                                "tool_call_truncated",
+                                {
+                                    "reason": truncation_reason,
+                                    "has_tool_calls": True,
+                                },
+                            )
+                            raise ToolCallTruncatedError(
+                                reason=truncation_reason, has_tool_calls=True
+                            )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)